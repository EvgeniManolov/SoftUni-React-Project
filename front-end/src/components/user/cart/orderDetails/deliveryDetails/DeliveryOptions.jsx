import React from 'react';

import { FormGroup, Radio } from 'react-bootstrap';

import { CART } from '../../../../../data/constants/componentConstants';


class DeliveryOptions extends React.Component {
	constructor (props) {
		super(props);

		this.state = {
			toAddress: this.props.toAddress,
		};
	}

	render () {
		return (

			<FormGroup className="delivery-options">
				<Radio
					name="1"
					inline
					defaultChecked={this.state.toAddress}
					onClick={(e) => {
						this.setState({toAddress: true}, () => {
							this.props.onChange('toAddress', this.state.toAddress);
						})
					}}>
					{CART.toAddress}
				</Radio>{' '}
			</FormGroup>
		);
	}
}

export default DeliveryOptions;
