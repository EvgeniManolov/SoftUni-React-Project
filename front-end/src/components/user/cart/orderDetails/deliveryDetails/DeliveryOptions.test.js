import React from 'react';
import DeliveryOptions from './DeliveryOptions';
import {shallow} from 'enzyme';

it('renders correctly', () => {
    const wrapper = shallow(<DeliveryOptions/>)
    expect(wrapper).toMatchSnapshot();
  });


  it('check for html changes', () => {
    const wrapper = shallow(<DeliveryOptions/>)
    expect(wrapper).toBeTruthy();
});