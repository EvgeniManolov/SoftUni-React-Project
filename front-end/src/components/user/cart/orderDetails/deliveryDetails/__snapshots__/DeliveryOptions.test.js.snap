// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DeliveryOptions />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "bsClass": "form-group",
      "children": Array [
        <Radio
          bsClass="radio"
          defaultChecked={undefined}
          disabled={false}
          inline={true}
          name="1"
          onClick={[Function]}
          title=""
>
          Доставка до адрес
</Radio>,
        " ",
      ],
      "className": "delivery-options",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "bsClass": "radio",
          "children": "Доставка до адрес",
          "defaultChecked": undefined,
          "disabled": false,
          "inline": true,
          "name": "1",
          "onClick": [Function],
          "title": "",
        },
        "ref": null,
        "rendered": "Доставка до адрес",
        "type": [Function],
      },
      " ",
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "bsClass": "form-group",
        "children": Array [
          <Radio
            bsClass="radio"
            defaultChecked={undefined}
            disabled={false}
            inline={true}
            name="1"
            onClick={[Function]}
            title=""
>
            Доставка до адрес
</Radio>,
          " ",
        ],
        "className": "delivery-options",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bsClass": "radio",
            "children": "Доставка до адрес",
            "defaultChecked": undefined,
            "disabled": false,
            "inline": true,
            "name": "1",
            "onClick": [Function],
            "title": "",
          },
          "ref": null,
          "rendered": "Доставка до адрес",
          "type": [Function],
        },
        " ",
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": false,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
